deps/raft/include/raft_private.h

```diff
@@ -118,8 +118,10 @@ typedef struct {
 
     raft_node_id_t node_transferring_leader_to; // the node we are targeting for leadership
     long transfer_leader_time; // how long we should wait for leadership transfer to take, before aborting
+    int sent_timeout_now; // if we've already sent a leadership transfer signal
 
     int timeout_now;
+    int timed_out;
```

Comment:
Are you trying to prevent second election with this variable? Can you describe why do we need it?

Reply:
yes.  its to not call election multiple times through, we can't reset timeout_now yet, as needed for the flag to requestvote.

Reply:
I was thinking about this, how about in raft_recv_requestvote(); we make this change :

```c
    /* Reject request if we have a leader */
    if (vr->prevote &&                                      << -- prevote check here
        !vr->transfer_leader &&                  
        me->leader_id != RAFT_NODE_ID_NONE &&
        me->leader_id != vr->candidate_id &&
        me->timeout_elapsed < me->election_timeout) {
        goto done;
    }
```

We can do leader stickiness test only for prevotes. So, in this way, we can avoid timed_out flag. 

```c

if ((me->election_timeout_rand <= me->timeout_elapsed || me->timeout_now &&
        !raft_snapshot_is_in_progress(me_))
    {
        int e = raft_election_start(me_);   /// me->timeout_now will be used for prevotes.
        if (0 != e)
            return e;
    }
   
    me->timeout_now = 0;  // We can clear timeout_now
```

Reply:
I'm not sure I agree with that logic change.  need to think it through.  i.e. I always viewed prevote as more of a connectivity check / "I could conceptually vote for you for leader, but not guaranteed", so the sticky leader is more important on the actual vote, not prevote.  but perhaps I'm wrong

Reply:
with that said, perhaps the way to do it is not have prevote handler ignore sticky, but return the "transfer leader" flag in response, and when candidate gets responses from prevote, it uses that to send the right transfer_leader flag.  though, this would be a bit more invasive.  right now raft_send_requestvote() sets the transfer_leader flag based on me->timeout_now, what I'm proposing is that send_requestvote() func would take an extra arg that is transfer_leader and uses that to set the structs transfer_leader member.

so raft_become_candidate() will take it as a flag also (from raft_recv_requestvote_response) so it can pass it to raft_send_requestvote and raft_become_precandidate() will take it from me->timeout_now.

my only Q would be do we need some sort of assert in raft_recv_requestvote_response to ensure that all responses are transfer_leader or not or can we rely on just last one being accurate?

Reply:
ok, I'm going to add a commit that is more complicated as I described above, but can go with your idea if you think its better.

Reply:
It should work, hopefully :) I don't see a problematic scenario and I just find this better because it might be the smallest change among other options. No strong opinion though.